Core/Model.py

```diff
@@ -296,3 +295,77 @@ def generate_direct_transition_system(self, max_time: float = np.inf, max_size:
             time.sleep(1)
 
         return ts
+    
+    def export_sbml(self) -> libsbml.SBMLDocument:
+        """
+        Convert model to a SBML model using SBML-multi package (libsbml).
+        :return: SBML document
+        """
+        test_model = ModelSBML()
+        test_model.docPlug.setRequired(True)
+        unique_complexes, unique_params = self.create_unique_complexes_and_params()
+
+        test_model.create_basic_species_types(self.atomic_signature, self.structure_signature) #1
+        test_model.create_all_species_compartments_and_complex_species_types(unique_complexes)#2
+        test_model.create_all_reactions(self.rules)#3
+        test_model.create_reaction_for_isomorphisms(unique_complexes)#3.1
+        test_model.create_parameters(self.definitions, unique_params)#4
+        test_model.set_initial_amounts(self.init)#5
+
+        return test_model.document
+
+    def create_unique_complexes_and_params(self):
+        """
+        Extracts unique complexes and compartments from rules
+        splits complexes by its isomorphisms
+        :return: dict of unique complexes mapped to list of its isomorphisms and set of unique params
+        """
+        #here all complexes will be (complexAgent:{(complexAgent, SBML_code_isomprphism1),(complexAgent, SBML_code_isomprphism2)... })
+        unique_complexes = dict()
+        #gets all initialization complexes
+        initialization_complexes = set(map(lambda x : x[0],self.init.items()))
+        unique_params_from_rate = set() # Might be usefull later
+
+        #traverses all rules
+        for rule in self.rules:
+            # gets all params and agents from single rate
+            agents, params = rule.rate.get_params_and_agents()
+
+            #get dict of all complex agents in rule as 'Keys'
+            #mapped to all occurences of isomorphism of this agent
+            #as SBML_code as 'Values' in set of tuples (compAgent, specific_SBML_code)
+            complexes_from_rule_in_dict = rule.get_unique_complexes_from_rule()
+            #this adds all elements in dict rule to all
+            #complexes from previous runs
+            for comp in complexes_from_rule_in_dict:
+                if comp not in unique_complexes:
+                    unique_complexes[comp] = complexes_from_rule_in_dict[comp]
+                else:
+                    unique_complexes[comp] = unique_complexes[comp].union(complexes_from_rule_in_dict[comp])
+            #this adds all agents from rates
+            #to all complexes from previous runs
+            for agent in agents:
+                double_agent = (agent, agent.to_SBML_species_code())
+                if agent not in unique_complexes:
+                    empty_set = set()
+                    empty_set.add(double_agent)
+                    unique_complexes[agent] = empty_set
+                else:
+                    unique_complexes[agent].add(double_agent)
```

Comment:
This code is basically repeated above for `complexes_from_rule_in_dict` and below for `initialization_complexes`. I think it could be generalised and extracted to a function.

Also instead of such condition, it can be written in a simpler way using `dict.get` method:

```
unique_complexes[agent] = unique_complexes.get(agent, set()) | {double_agent}
```

where `get` returns value for `agent` if exists, else defaults to `set()`. In both cases, you can just union the `double_agent` to that and assign to the dict.